ibis/tests/expr/test_value_exprs.py

```diff
@@ -1213,17 +1211,16 @@ def test_map_get_with_compatible_value_bigger():
 
 def test_map_get_with_incompatible_value_different_kind():
     value = ibis.literal({'A': 1000, 'B': 2000})
-    with pytest.raises(IbisTypeError):
-        value.get('C', 3.0)
+    assert value.get('C', 3.0).type() == dt.float64
```

Comment:
I changed the semantics here to use the highest precedence one.

Reply:
We should try this on at least two backends to see if in practice this behavior works.

Reply:
Addressed it in a follow-up PR https://github.com/ibis-project/ibis/pull/4476